### PR TITLE
Payment widget

### PR DESCRIPTION
**Pull request by ratik21**
image
image
image
image
image

### CODE REVIEW

**Comment by ratik21** [comment]
Note: import from `@algo-builder/web` is not working, added TODO in code.

File: shop/src/components/payment-widget.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  if (walletAccounts.length) {
+    return walletAccounts[0].address;
+  }
+  return undefined;
+}
+
+/**
+ * Executes transaction to receive payment of ticket
+ * @param fromAddress address of currently active account in wallet
+ * @param {*} toAddress address to recieve payment (currently set to master-account)
```
Comment (amityadav0) [inline shop/src/components/payment-widget.js]: what is `{*}` ?

File: shop/src/components/payment-widget.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    return ['Error Occurred: ' + error.message];
+  }
+}
+
+/**
+ * Widget to trigger a purchase on click. Consists of a Button, Loading State
+ * and a text prop to show below button (where text is the transaction response)
+ * @param buttonText text to display on button
+ * @amount amount(of ticket) in ALGOs to charge user
+ *
```
Comment (amityadav0) [inline shop/src/components/payment-widget.js]:
```suggestion
```

File: shop/src/components/payment-widget.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+ */
+export const PaymentWidget = ({ buttonText, amount }) => {
+  const [result, setResult] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  const executeTx = useCallback(async () => {
+    // This is the master account address present in algob.config.js
+    // We will take payments in this address
+    const toAddress =
+      'WWYNX3TKQYVEREVSW6QQP3SXSFOCE3SKUSEIVJ7YAGUPEACNI5UGI4DZCE';
```
Comment (amityadav0) [inline shop/src/components/payment-widget.js]:
```suggestion
    const toAddress = 'WWYNX3TKQYVEREVSW6QQP3SXSFOCE3SKUSEIVJ7YAGUPEACNI5UGI4DZCE';
```

File: shop/src/components/payment-widget.js (hunk 1 above)
Comment (ratik21) [reply shop/src/components/payment-widget.js]: used to represent  type information. removing

**Review by amityadav0 — APPROVED** [review]

**Pull request merged**